drivers/gpio/gpio_gd32.c

```diff
@@ -286,43 +286,53 @@ static int gpio_gd32_pin_interrupt_configure(const struct device *port,
 					     enum gpio_int_mode mode,
 					     enum gpio_int_trig trig)
 {
-	if (mode == GPIO_INT_MODE_DISABLED) {
-		gd32_exti_disable(pin);
-		(void)gd32_exti_configure(pin, NULL, NULL);
-		gd32_exti_trigger(pin, GD32_EXTI_TRIG_NONE);
-	} else if (mode == GPIO_INT_MODE_EDGE) {
-		int ret;
-
-		ret = gd32_exti_configure(pin, gpio_gd32_isr, (void *)port);
-		if (ret < 0) {
-			return ret;
-		}
+	/*
+	 * The if-else clause intends to eliminate dependency on EXTI.
+	 * For suppressing code generation of the else part at compile time,
+	 * the condition needs to determine statically.
+	 */
+	if (IS_ENABLED(CONFIG_GD32_EXTI)) {
+		if (mode == GPIO_INT_MODE_DISABLED) {
```

Comment:
IMO, what should be done is the following:

1. Add a Kconfig option to build the GD32 driver w/o interrupt support, e.g. `CONFIG_GPIO_GD32_INT_ENABLE`
2. Such option should be enabled by default, and select `GD32_EXTI` (intc drivers do not have a class, so we need to select on a driver basis)
3. Just ifdefout code if `CONFIG_GPIO_GD32_INT_ENABLE=n`. I've personally never liked `IS_ENABLED` because it can't be universally used without getting compile errors in some cases.

But still, I feel it is wrong to add such option at a driver level. It should likely be a subsystem setting to build GPIO drivers with/without interrupt support.

Reply:
It is as you say. I look at the implementation of other GPIO drivers, and none of them enable or disable .pin_interrupt_configure in the settings. So I would like to withdraw this idea.
However, I think that it is better to clarify the dependency on EXTI,
I have changed the fix (and PR title).
Could you please confirm this?